src/middlewares/checkTokenMiddleware.ts

```diff
@@ -0,0 +1,21 @@
+import { NextFunction, Request, Response } from 'express';
+import * as jwt from 'jsonwebtoken';
+
+export const checkTokenMiddleware = function (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) {
+  const token = req.headers['jwt-access-token'] as string;
```

Comment:
Look at the types of req.headers and consider all possibilities (e.g. if it's string, if it's array of string, or if it's undefined).

Reply:
fixed